### PR TITLE
Add option to install Botkube via CLI, add `--verbose` flag

### PR DESCRIPTION
**Pull request by mszostok**
## Description

Changes proposed in this pull request:

- Add option to install Botkube via CLI
      - use Helm Go SDK underneath to install the chart
      - make sure it can be run with debug/verbose
      - support dry-run
      - support Botkube version and overrides
      - Respect KUBECONFIG env variable
- Add `--verbose` flag
  - it supports also modes that can be used to print more details. For example, for GraphQL call, we can print requests/responses when `--verbose=trace` is used. It should be useful to print it only if needed, as printing that all the time could be too messy.

## Testing

1. Build the CLI
2. Create k8s cluster
3. Run:

```bash
export SLACK_CHANNEL_NAME=random
export SLACK_APP_TOKEN="xapp-1-"
export SLACK_BOT_TOKEN="xoxb-"

botkube install --verbose \
--set communications.default-group.socketSlack.enabled=true \
--set communications.default-group.socketSlack.channels.default.name=${SLACK_CHANNEL_NAME} \
--set communications.default-group.socketSlack.appToken=${SLACK_API_APP_TOKEN} \
--set communications.default-group.socketSlack.botToken=${SLACK_API_BOT_TOKEN} \
--set settings.clusterName=labs \
--set 'executors.k8s-default-tools.botkube/kubectl.enabled'=true \
--set 'executors.k8s-default-tools.botkube/helm.enabled'=true
```

here are described all `install` flags: https://github.com/kubeshop/botkube/blob/db6a62fcce7dee3856827fa05a32971e837c400f/cmd/cli/docs/botkube_install.md

https://github.com/kubeshop/botkube/assets/17568639/c61ff2f3-3d6f-4a8f-9ac4-23be107c9eca

## Notes

**In the next PR I will add streaming Botkube Pod logs. This PR is already huge and I'm still working on logs.**

## Related issue(s)

- https://github.com/kubeshop/botkube/issues/1133

### CODE REVIEW

File: internal/kubex/config.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+	// 3. In-cluster config if running in cluster
+	// 4. $HOME/.kube/config if exists
+	// 5. user.HomeDir/.kube/config if exists
+	//
+	// NOTE: For default config file locations, upstream only checks
+	// $HOME for the user's home directory, but we can also try
+	// os/user.HomeDir when $HOME is unset.
+	//
+	// TODO(jlanford): could this be done upstream?
```
Comment (huseyinbabal) [inline internal/kubex/config.go:63]: `jlanford`? :)

File: internal/cli/install/install.go [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+	renderer := style.NewGoTemplateRender(style.DefaultConfig(successInstallGoTpl))
+
+	props := map[string]string{
+		"DocsURL": "https://docs.botkube.io/cli/getting-started/#first-use",
+		"Version": version,
+	}
+
+	out, err := renderer.Render(props, cli.IsSmartTerminal(w))
+	if err != nil {
+		return err
```
Comment (huseyinbabal) [inline internal/cli/install/install.go:93]: we can add err context

**Review by huseyinbabal — APPROVED** [review]
Lot's of improvements, thank you 🚀

File: internal/kubex/config.go (hunk 1 above)
Comment (mszostok) [reply internal/kubex/config.go:63]: as mentioned in the function comment, I copied that from controller-runtime and I wanted to keep is as close as possible to original. But I can remove that comment 👍

**Pull request merged**